### PR TITLE
Desynchronize viewport from the IC chat log so that users have a smooth viewing experience as an option

### PR DESCRIPTION
**Pull request by Crystalwarrior**
This PR also completely fuckin overhauls handle_chatmessage monstrosity and unspaghettifies code

TODO:
 - [x] Create a queue that contains chat msg packets and goes through them one by one
 - [x] The chat message needs to be logged as soon as it is received however, making it outside of the queue parser
 - [x] The queue will go through each message once the previous message has finished
 - [x] Interjections are intentionally delayed slightly so they don't cut off messages instantly if queued.
 - [x] A settings option to either display IC logs as they're received, or to delay the IC logs until the message is reached
 - [x] There should be a setting for minimum wait in milliseconds between messages
 - [ ] Indicate currently parsed viewport message by highlighting it in the IC chat log in some way
 - [ ] If your own message has been sent, and Desynchronized Logs is disabled, make it still appear but greyed out (transparent) so that you're at least aware your message is still yet to be processed.

### CODE REVIEW

**Review by in1tiate — DISMISSED** [review]
i enjoy this immensely

File: include/courtroom.h [hunk 1]
```diff
@@ -305,6 +340,8 @@ class Courtroom : public QMainWindow {
 
   QVector<chatlogpiece> ic_chatlog_history;
 
+  QQueue<QStringList> chatmessage_queue;
```
Comment (in1tiate) [inline include/courtroom.h:343]: huh what what's a QQueue

Comment (Crystalwarrior) [reply include/courtroom.h:343]: its a queue :^)

**Review by in1tiate — APPROVED** [review]
aye

**Pull request merged**